### PR TITLE
Simplify ContentView: extract DocumentSurfaceViewModel (closes #323)

### LINKED ISSUE #323: Simplify ContentView: extract document surface orchestration from view layer
## Problem

`ContentView` violates SRP by mixing view rendering with business logic. It owns 7 controller references, manages two web surface modes (preview/source), drop targeting, split scroll coordination, change navigation, source HTML caching, and document surface layout — all in a single SwiftUI `View`.

Views should compose UI, not orchestrate behavior. The current structure makes the view hard to test, hard to preview, and fragile to modify.

## Current Responsibilities (all in one View)

1. **Document surface mode management** — `PreviewMode` (web/nativeFallback) and `SourceMode` (web/plainTextFallback) enums with state transitions
2. **Surface configuration building** — `documentSurfaceConfiguration(for:)`, `documentSurfacePane(for:)` with complex conditional logic
3. **Source HTML caching** — `SourceHTMLDocumentCache` invalidation, `refreshSourceHTML()`, `sourceDocumentIdentity`
4. **Drop targeting** — `DropTargetingCoordinator` state, `handleDroppedFileURLs`, `canAcceptDroppedFileURLs`
5. **Split scroll coordination** — `SplitScrollCoordinator`, `handleScrollSyncObservation`, `splitScrollRequest`
6. **Changed region navigation** — `ChangedRegionNavigationCoordinator`, `canNavigateChangedRegions`
7. **Overlay layout computation** — `overlayInsets`, `overlayTopInset`, `isStatusBannerVisible`, loading overlay state
8. **File identity change handling** — `handleFileIdentityChange`, `handleSurfaceAppear`, `promptForImageDirectoryAccess`
9. **Mode/view transitions** — `handlePreviewModeChange`, `handleSourceModeChange`, `handleDocumentViewModeChange`

### A. `DocumentSurfaceViewModel` (new `@Observable` controller)
Move the orchestration logic out of the view:
- **Mode state** — `PreviewMode`, `SourceMode`, and transitions between them
- **Surface configuration building** — `documentSurfaceConfiguration(for:)`, `documentSurfacePane(for:)`
- **File identity handling** — `handleFileIdentityChange`, `handleSurfaceAppear`
- **Image access prompting** — `promptForImageDirectoryAccess`

The view becomes a thin composition layer that reads from this view model.

### B. Move source HTML caching into the rendering pipeline
- `SourceHTMLDocumentCache` invalidation and `refreshSourceHTML()` belong in `ReaderRenderingController` (which already owns rendering state). The view should observe rendered output, not manage cache invalidation.

### C. Split into 3 focused views
Rather than one `ContentView` with conditionals:
- `PreviewDocumentSurface` — handles `PreviewMode` (web vs native fallback)
- `SourceDocumentSurface` — handles `SourceMode` (web vs plain text fallback)  
- `DocumentSurfaceContainer` — owns `DocumentSurfaceLayoutView`, mode switching, overlay coordination

Each surface view owns its own web view configuration and fallback logic.

### D. Keep thin in ContentView
After extraction, `ContentView` should only:
- Read from `DocumentSurfaceViewModel`
- Compose the surface views with overlay views (watch pill, change navigation, TOC, utility rail)
- Forward user actions via `onAction` callback

## Approach

- Start with extracting `DocumentSurfaceViewModel` — this is the highest-value, lowest-risk step.
- Move source HTML cache invalidation into `ReaderRenderingController` as a second step.
- Split the surface views last, as this touches the view hierarchy.
- All steps are purely structural with no behavior changes.
- The existing controller pattern (`ReaderDocumentController`, `ReaderRenderingController`, etc.) provides the model for how the new view model should look.

## Benefits

- ContentView shrinks from ~50 computed properties/methods to ~15.
- Document surface logic becomes unit-testable via the view model.
- Preview and source surfaces can evolve independently.
- Easier to add new surface modes (e.g., a diff-only view) without touching the container.

### PR DESCRIPTION
**Pull request by larspohlmann**
## Summary

Extracts document surface orchestration out of `ContentView` into a dedicated `@Observable` `DocumentSurfaceViewModel`, addressing #323.

The view model owns mode state (`PreviewMode`/`SourceMode`), reload tokens, shared action dispatch, surface-configuration building, source HTML caching, and coordinator handles (split scroll, drop targeting, change navigation). `ContentView` is now a composition layer that reads from the view model and forwards user actions. `ContentViewConfigurationFactory` has been removed (its logic moved into the view model). `SplitScrollCoordinator` is extracted into its own file.

The view model is owned via `@State` on `ContentViewAdapter` so it survives observable-driven body re-evaluations — previously a local `let` in `body`, which would have wiped fallback modes, in-flight scroll sync, the change navigation index, drop state, and the source HTML cache on every observable change.

## Changes

- New: `DocumentSurfaceViewModel` (`@MainActor` / `@Observable`) with full test coverage in `DocumentSurfaceViewModelTests`.
- New: `SplitScrollCoordinator.swift` (extracted from its prior location).
- Removed: `ContentViewConfigurationFactory.swift` (logic absorbed by the view model).
- `ContentView.swift` trimmed from a grab-bag of orchestration to composition + overlays; `canNavigateChangedRegions` consolidated into a single computed property (was evaluated three times per body).
- `ContentViewAdapter` gains an explicit `init` that wraps the view model in `State(wrappedValue:)` so its state persists.

## Test plan

- [x] `xcodebuild build` passes (macOS, Debug).
- [x] Full `minimarkTests` suite passes.
- [x] New `DocumentSurfaceViewModelTests` cover mode transitions, source-mode fallback edge cases, identity generation, shared action dispatch, and configuration building.
- [ ] Manual smoke: open a document, toggle preview/split/source modes, drop a file, trigger a preview crash fallback, navigate changed regions, confirm split scroll sync still tracks.

Closes #323.

### CODE REVIEW

File: minimark/Views/Content/DocumentSurfaceViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+            && previewMode == .web
+            && sourceMode == .web
+    }
+
+    func documentSurfaceConfiguration(
+        for surface: DocumentSurfaceRole,
+        fileURL: URL?,
+        renderedHTMLDocument: String,
+        sourceMarkdown: String,
+        documentViewMode: ReaderDocumentViewMode,
```
[the remark this answers is not in the document]
Comment (larspohlmann) [reply minimark/Views/Content/DocumentSurfaceViewModel.swift:155]: Fixed in f0e4a47 — dropped the unused `sourceMarkdown` parameter from `documentSurfaceConfiguration`.

File: minimark/Views/Content/DocumentSurfaceViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+enum SourceMode {
+    case web
+    case plainTextFallback
+}
+
+@MainActor
+@Observable
+final class DocumentSurfaceViewModel {
+
```
[the remark this answers is not in the document]
Comment (larspohlmann) [reply minimark/Views/Content/DocumentSurfaceViewModel.swift]: Fixed in f0e4a47 — nested `PreviewMode` and `SourceMode` under `DocumentSurfaceViewModel` as suggested.

File: minimark/ContentView.swift [hunk 3]
```diff
@@ -518,6 +460,86 @@ struct ContentView: View {
...
+    private func recoverFallbackSurfaceModes() {
+        if surfaceViewModel.previewMode == .nativeFallback, !rendering.renderedHTMLDocument.isEmpty {
+            surfaceViewModel.previewReloadToken += 1
+            surfaceViewModel.previewMode = .web
+        }
+        if surfaceViewModel.sourceMode == .plainTextFallback,
+           !document.sourceMarkdown.isEmpty {
+            surfaceViewModel.sourceReloadToken += 1
+            surfaceViewModel.sourceMode = .web
+        }
```
[the remark this answers is not in the document]
Comment (larspohlmann) [reply minimark/ContentView.swift]: Fixed in f0e4a47. Removed `recoverFallbackSurfaceModes()` from ContentView and rewrote `DocumentSurfaceViewModel.handleSurfaceAppear(renderedHTMLDocument:sourceMarkdown:)` to take the inputs directly — onAppear now calls that single entry point. Also dropped the dead `renderedHTMLDocumentProvider`/`sourceMarkdownProvider` fields and the provider-closure init. Note: the source HTML refresh stays separate (it needs `settingsStore.currentSettings`, not a stale closure) and continues to flow through `refreshSourceHTML`.

File: minimark/Views/Content/DocumentSurfaceViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+    var previewMode: PreviewMode = .web
+    var sourceMode: SourceMode = .web
+    var previewReloadToken = 0
+    var sourceReloadToken = 0
+
+    let splitScrollCoordinator = SplitScrollCoordinator()
+    var dropTargeting = DropTargetingCoordinator()
+    var changeNavigation = ChangedRegionNavigationCoordinator()
+    var sourceHTMLCache = SourceHTMLDocumentCache()
```
[the remark this answers is not in the document]
Comment (larspohlmann) [reply minimark/Views/Content/DocumentSurfaceViewModel.swift:26]: Good catch — this was the exact bug. Fixed in f930831 (pushed before your review was posted): converted `SplitScrollCoordinator` to `@Observable` and dropped `ObservableObject`/`@Published`. Added `scrollSyncObservation_invalidatesObservers` as a regression test that asserts mutations on the coordinator invalidate observers of `documentSurfaceConfiguration`.
